### ISSUE TITLE
Support for custom headers

### ISSUE THREAD

**Issue by codekoala** [hashicorp/go-getter#71]
I want to download build artifacts from a private GitLab instance using Nomad's artifact stanza. I can download the build artifacts with curl if I add a ``Private-Token`` header, but there doesn't seem to be a way to handle with ``go-getter`` right now.

**Comment by lfarnell** [comment]
@codekoala the [docs](https://docs.gitlab.com/ee/api/README.html#basic-usage) say you can pass the private token via a query parameter as well. It's not secure but it should work. Not sure how much would have to change for go-getter and nomad to support custom headers

**Comment by codekoala** [comment]
@lfarnell thank you for the response! It's been a while since I worked on this (ended up pushing artifacts to S3), but I'm *fairly* certain I tried that query string parameter without results. I think it worked for certain part of the API, but the artifacts seemed to be an exception for everything other than the header. I will try the query parameter later tonight and report back. That would be great if it works!

One way or another, being able to specify custom headers seems like a good feature to deal with less flexible APIs.
